### ISSUE TITLE
Inherited functions can't be overridden if called from an inherited event

### ISSUE THREAD

**Issue by protogenposting** [YoYoGames/GameMaker-Bugs#8434]
### Description

This is a really freaky bug.

basically, calling a function in an inherited event will call the function that was originally created in the inherited event. Idk how to explain it better

https://api.gamemaker.io/api/github/downloads/893e65d2-7206-4da9-91fb-942f025038ae

here's a small test project

[Override test.zip](https://github.com/user-attachments/files/17787081/Override.test.zip)

### Steps To Reproduce

1. Start GameMaker
2. Load the test project
3. Run it
4. press space
5. See the issue

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.1100 (Betas)

### Which platform(s) are you seeing the problem on?

Ubuntu

### Sample Package Attached?

- [x] I have attached my bug package

### Sample Project Added?

- [x] I have included a small sample project

**Comment by protogenposting** [comment]
apparently I wasn't on the latest beta. Updating to the new runtime did not fix the issue though

**Comment by Notlea** [comment]
Got the same issue. In my case I have an empty state declared in the parent object that is being overwritten in the child. When the child performs a call to a new function, it actually invokes empty function from the parent. I have to mention, that empty function is assigned at the parent, not in the child. 
Parent:
```
//create event
state_idle = function() {
};
state = state_idle;
    //step event
state();
```
Child:
```
state_idle = function() {
    //some logic to perform
};
```
I solved my issue by explicitly assigning state in the child:
Child:
```
//create event
state_idle = function() {
    //some logic to perform
};
state = state_idle; //reassignment
//step event is still inherited
```

I'm using IDE 2024.6.2.162 with Runtime 2024.6.1.208

**Comment by rwkay** [comment]
Fixed in 2024.11 - not sure what the issue is in the above comment but the bug here was only present in the 2024.11 Beta

**Comment by Emc1923** [comment]
@rwkay will this fix be implemented for the HMLT5 target as well? Thank you.

**Comment by Emc1923** [comment]
Reopening to get the fix added for HTML5

**Comment by rwkay** [comment]
Fixed in 2024.11

**Comment by Emc1923** [comment]
Verified fixed in  IDE v2024.1100.0.699  Runtime v2024.1100.0.717.
